### PR TITLE
Add Mighty Mike 3.0.1

### PR DESCRIPTION
**Pull request by ivanholmes**
Incorporating patches from the Nanosaur port #6891

### CODE REVIEW

**Comment by ivanholmes** [comment]
Ready to merge from my POV.

File: games-action/mightymike/mightymike-3.0.1.recipe [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+CHECKSUM_SHA256="10b1cfc90e601b0bd875fd959127f1eaf1ffbc0ad54aa03db5f86197e6b2c8b9"
+SOURCE_DIR="MightyMike-$portVersion"
+PATCHES="mightymike-$portVersion.patchset"
+ADDITIONAL_FILES="mightymike.rdef.in"
+srcGitRev2="0581b241624d16c453e0afab57a379b443229260"
+SOURCE_URI_2="https://github.com/jorio/Pomme/archive/$srcGitRev2.tar.gz"
+CHECKSUM_SHA256_2="5e74b0061908665e7a9c710d83969aa0234b78766b05eacfa86cc3c1c0d1f666"
+SOURCE_FILENAME_2="Pomme-$srcGitRev2.tar.gz"
+SOURCE_DIR_2="Pomme-$srcGitRev2"
+PATCHES_2="mightymike-$portVersion-source2.patchset"
```
Comment (korli) [inline games-action/mightymike/mightymike-3.0.1.recipe:24]:
```suggestion
SOURCE_URI="https://github.com/jorio/MightyMike/archive/refs/tags/v$portVersion.tar.gz"
CHECKSUM_SHA256="10b1cfc90e601b0bd875fd959127f1eaf1ffbc0ad54aa03db5f86197e6b2c8b9"
SOURCE_DIR="MightyMike-$portVersion"
PATCHES="mightymike-$portVersion.patchset"
srcGitRev2="0581b241624d16c453e0afab57a379b443229260"
SOURCE_URI_2="https://github.com/jorio/Pomme/archive/$srcGitRev2.tar.gz"
CHECKSUM_SHA256_2="5e74b0061908665e7a9c710d83969aa0234b78766b05eacfa86cc3c1c0d1f666"
SOURCE_FILENAME_2="Pomme-$srcGitRev2.tar.gz"
SOURCE_DIR_2="Pomme-$srcGitRev2"
PATCHES_2="mightymike-$portVersion-source2.patchset"
ADDITIONAL_FILES="mightymike.rdef.in"
```

File: games-action/mightymike/mightymike-3.0.1.recipe [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	mightymike$secondaryArchSuffix = $portVersion
+	app:MightyMike = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libsdl2_2.0$secondaryArchSuffix
+	lib:libgl$secondaryArchSuffix
```
Comment (korli) [inline games-action/mightymike/mightymike-3.0.1.recipe:36]:
```suggestion
	lib:libgl$secondaryArchSuffix
	lib:libsdl2_2.0$secondaryArchSuffix
```

File: games-action/mightymike/mightymike-3.0.1.recipe [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libsdl2_2.0$secondaryArchSuffix
+	devel:libgl$secondaryArchSuffix
```
Comment (korli) [inline games-action/mightymike/mightymike-3.0.1.recipe:42]:
```suggestion
	devel:libgl$secondaryArchSuffix
	devel:libsdl2_2.0$secondaryArchSuffix
```

File: games-action/mightymike/mightymike-3.0.1.recipe [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:make
+	cmd:gcc$secondaryArchSuffix
```
Comment (korli) [inline games-action/mightymike/mightymike-3.0.1.recipe:47]:
```suggestion
	cmd:gcc$secondaryArchSuffix
	cmd:make
```

File: games-action/mightymike/mightymike-3.0.1.recipe [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	"
+
+BUILD()
+{
+	# submodule
+	rm -rf extern/Pomme
+	ln -s $sourceDir2 extern/Pomme
+	# building
+	cmake -S . -B build -DCMAKE_BUILD_TYPE=RelWithDebInfo
```
Comment (korli) [inline games-action/mightymike/mightymike-3.0.1.recipe:56]:
```suggestion
	cmake -S . -B build -DCMAKE_BUILD_TYPE=Release
```

**Pull request merged**